src/Modules/Visualization/ShowOrientationAxes.cc

```diff
@@ -0,0 +1,182 @@
+{
+  x_ = x;
+  y_ = y;
+  z_ = z;
+}
+
+void ShowOrientationAxesImpl::setScale(double scale)
+{
+  scale_ = scale;
+}
+
+RenderState ShowOrientationAxesImpl::getRenderState() const
+{
+  RenderState renState;
+  renState.set(RenderState::USE_NORMALS, true);
+  renState.set(RenderState::IS_ON, true);
+  renState.set(RenderState::USE_TRANSPARENCY, false);
+  renState.mGlyphType = RenderState::GlyphType::ARROW_GLYPH;
+  renState.set(RenderState::USE_DEFAULT_COLOR, false);
+  return renState;
+}
+
+void ShowOrientationAxesImpl::addOrientationArrows(GlyphGeom& glyphs, const Point& pos) const
+{
+  double radius = 0.1 * scale_;
+
+  // Positive
+  glyphs.addArrow(pos, pos + Vector(scale_, 0, 0), radius, ARROW_RATIO_, RESOLUTION_,
+                  RED_, RED_, RENDER_CYLINDER_BASE_, RENDER_CONE_BASE_);
+  glyphs.addArrow(pos, pos + Vector(0, scale_, 0), radius, ARROW_RATIO_, RESOLUTION_,
+                  GREEN_, GREEN_, RENDER_CYLINDER_BASE_, RENDER_CONE_BASE_);
+  glyphs.addArrow(pos, pos + Vector(0, 0, scale_), radius, ARROW_RATIO_, RESOLUTION_,
+                  BLUE_, BLUE_, RENDER_CYLINDER_BASE_, RENDER_CONE_BASE_);
+
+  // Negative
+  glyphs.addArrow(pos, pos - Vector(scale_, 0, 0), radius, ARROW_RATIO_, RESOLUTION_,
+                  RED_NEGATIVE_, RED_NEGATIVE_, RENDER_CYLINDER_BASE_, RENDER_CONE_BASE_);
+  glyphs.addArrow(pos, pos - Vector(0, scale_, 0), radius, ARROW_RATIO_, RESOLUTION_,
+                  GREEN_NEGATIVE_, GREEN_NEGATIVE_, RENDER_CYLINDER_BASE_, RENDER_CONE_BASE_);
+  glyphs.addArrow(pos, pos - Vector(0, 0, scale_), radius, ARROW_RATIO_, RESOLUTION_,
+                  BLUE_NEGATIVE_, BLUE_NEGATIVE_, RENDER_CYLINDER_BASE_, RENDER_CONE_BASE_);
+}
+
+BBox ShowOrientationAxesImpl::getBBox(const Point& pos) const
+{
+  BBox bbox = BBox();
+  Vector halfDiag = Vector(scale_, scale_, scale_);
+  bbox.extend(pos - halfDiag);
+  bbox.extend(pos + halfDiag);
+  return bbox;
+}
+
+GeometryHandle ShowOrientationAxesImpl::makeGeometry(const GeometryIDGenerator& idGen) const
+{
+  GlyphGeom glyphs;
+  Point pos = Point(x_, y_, z_);
+
+  addOrientationArrows(glyphs, pos);
+  BBox bbox = getBBox(pos);
+
+  auto geom(boost::make_shared<GeometryObjectSpire>(idGen, "ShowOrientationAxes", true));
+  glyphs.buildObject(*geom, geom->uniqueID(), false, 1.0, ColorScheme::COLOR_IN_SITU,
+                     getRenderState(), SpireIBO::PRIMITIVE::TRIANGLES, bbox, true, nullptr);
+  return geom;
+}
+
+ShowOrientationAxes::ShowOrientationAxes() : GeometryGeneratingModule(staticInfo_),
+                                             impl_(new ShowOrientationAxesImpl)
+{
+  INITIALIZE_PORT(InputField);
+  INITIALIZE_PORT(OutputGeom);
+}
+
+void ShowOrientationAxes::setStateDefaults()
+{
+  auto state = get_state();
+  state->setValue(Scale, 1.0);
+  state->setValue(X, 0.0);
+  state->setValue(Y, 0.0);
+  state->setValue(Z, 0.0);
+}
+
+void ShowOrientationAxes::execute()
+{
+  auto input = getRequiredInput(InputField);
```

Comment:
@RubioJr9 Here's the required input port...did you merge this funny or something? Or forget about it...

Reply:
Or maybe just leftover copy-paste...this is why the test network is useful 😄

Reply:
I have no idea where that came from. I tested it before the merge and I wasn't getting any errors about a required input port.

Reply:
That field input port is also in the module header file, is it supposed to be an optional input? Seems unlikely it would appear in both files by mistake.

Reply:
Yeah I must have copied it over by accident. I'll remove it now